### PR TITLE
Conan Package Manager support

### PR DESCRIPTION
**Pull request by vmalia**
# Description
PR adds Conan Package Manager support to the project to download and install many dependencies automatically.

Motivation: Better user experience when resolving dependencies.

Changes:
1. Added Conan related files.
2. Updated README & documentation.
3. Updated build system to support Conan build handle.
4. Added & updated Legal notices since Conan will download third-party software.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -19,6 +19,14 @@
 
 cmake_minimum_required (VERSION 3.13)
 
+project(oneMKL VERSION 0.1.0 LANGUAGES CXX)
+
```
Comment (mkrainiuk) [inline CMakeLists.txt:23]: For windows project should be defined after default parameters are set (it happens on line 41). We don't need it here

File: README.md [hunk 2]
```diff
@@ -253,7 +269,131 @@ Python | 3.6 or higher | [PSF](https://docs.python.org/3.6/license.html)
...
-3. [Build with CMake](#building-with-cmake).
+3. You can [Build with Conan](#building-with-conan) to automate the process of getting dependencies or you can download and install the required dependencies manually and [Build with CMake](#building-with-cmake) directly .
+
+*Note: Conan package manager automates the process of getting required packages, so that you do not have to go to different web location and follow different instructions to install them.*
+
+---
+
+## Building with Conan
+
+** Make sure you have completed [Build Setup](#build-setup). **
```
Comment (mkrainiuk) [inline README.md:280]: I guess we need to specify somewhere that conan doesn't support Windows build so far

File: CMakeLists.txt (hunk 1 above)
Comment (vmalia) [reply CMakeLists.txt]: Removed

File: README.md (hunk 2 above)
Comment (vmalia) [reply README.md:282]: Added a note.

**Comment by NadyaTen** [comment]
> PR adds Conan Package Manager support to the project to download and install many dependencies automatically.

What exactly "many dependencies" means?

File: conanfile.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+(the “Software Package”), and any included software or software made available \n \
+for download, you agree to the terms and conditions of the software license \n \
+agreements for the Software Package, which may also include notices, \n \
+disclaimers or license terms for third party software (the “Agreements”) \n \
+located at https://github.com/conan-io/conan/blob/develop/LICENSE.md, the \n \
+THIRD-PARTY-PROGRAMS file and in the README.md file included with the Software Package. \n \
+    "
+
+    # Dependencies
+    oneapi_version = "2021.1-beta05"
```
Comment (NadyaTen) [inline conanfile.py:41]: Shall we use beta06?

**Comment by vmalia** [comment]
> > PR adds Conan Package Manager support to the project to download and install many dependencies automatically.
> 
> What exactly "many dependencies" means?

I thought using "many dependencies" would be correct because it says - yes, Conan downloads "many" dependencies, but not "all" of them.
Please feel free to edit the PR description if you think something else would be better.

**Comment by vmalia** [comment]
> > > > PR adds Conan Package Manager support to the project to download and install many dependencies automatically.
> > > 
> > > 
> > > What exactly "many dependencies" means?
> > 
> > 
> > I thought using "many dependencies" would be correct because it says - yes, Conan downloads "many" dependencies, but not "all" of them.
> > Please feel free to edit the PR description if you think something else would be better.
> 
> I actually was wondering what exactly it will/can download?

The tables in README.md that mentions what Conan will download. You can check below tables:
1. https://github.com/vmalia/oneMKL/tree/pkgman3#hardware-and-os-specific
2. https://github.com/vmalia/oneMKL/tree/pkgman3#product-and-version-information

File: conanfile.py (hunk 3 above)
Comment (vmalia) [reply conanfile.py:41]: Let's have another PR for that change. Thank you.

**Pull request merged**